### ISSUE TITLE
Invariant_booleans false positive with while

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#415]
The 2 following `OK` are reported as lint.

```dart
void doWhile() {
  int value;
  if (value > 0) {
    do {
      value -= 1;
    } while (value > 0); // OK
  }
}

void while1() {
  int value;
  if (value > 0) {
    while (value > 0) { // OK
      value -= 1;
    }
  }
}
```

**Comment by dikmax** [comment]
I think it's not a bad thing to lint second one because it's same as without surrounding if.

```dart
void while1() {
  int value;
  while (value > 0) {
    value -= 1;
  }
}
```

**Comment by srawlins** [comment]
We've deprecated `invariant_booleans` and will not be fixing correctness bugs going forward.
